clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -1103,6 +1094,8 @@ shouldLifetimeExtendThroughPath(const IndirectLocalPath &Path) {
   for (auto Elem : Path) {
     if (Elem.Kind == IndirectLocalPathEntry::DefaultInit)
       return PathLifetimeKind::Extend;
+    if (Elem.Kind == IndirectLocalPathEntry::MemberExpr)
+      continue;
```

Comment:
Is this necessary for this case `const Bar& bar = foo.v.back(); // OK` ?

Reply:
This is necessary for the standard lifetime extension case. Since we add a new `MemberExpr` to the path, we need to skip it to maintain the existing behavior; otherwise, cases like `const int &x = A().i;` would incorrectly trigger diagnostics.

The case `const Bar& bar = foo.v.back();` is already handled on Line 1186.

Reply:
Got it. Thanks.